File: JavaScript/7js 2/rangeBitCount.js
/*
|--------------------------------------------------------------------------
| Simple Fun #10: Range Bit Counting
|--------------------------------------------------------------------------
|
*/

function rangeBitCount(a, b) {
   let res = "";
   for (let i = a; i <= b; i++) res += i.toString(2);
   return res.split("").filter((i) => i == 1).length;
}


function rangeBitCount(a, b) {
   return Array(b - a + 1)
      .fill(0)
      .map((_, i) => (i + a).toString(2).replace(/0/g, "").length)
      .reduce((s, n) => s + n, 0);
}


function rangeBitCount(a, b) {
   let sum = 0;
   for (let i = a; i <= b; i++) sum += i.toString(2).split("1").length - 1;
   return sum;
}


console.log(rangeBitCount(2, 7), 11);
console.log(rangeBitCount(0, 1), 1);
console.log(rangeBitCount(4, 4), 1);

File: Python/7py 2/range_bit_count.py
"""
|--------------------------------------------------------------------------
| Simple Fun #10: Range Bit Counting
|--------------------------------------------------------------------------
|
"""

def range_bit_count(a, b):
   return len(''.join(''.join([bin(i)[2:] for i in range(a,b+1)]).split('0')))


def range_bit_count(a,b):
   return sum(x.bit_count() for x in range(a,b+1))


def range_bit_count(a, b):
   return sum([sum(int(y) for y in bin(x)[2:] if y == '1') for x in range(a, b+1)])


def range_bit_count(a, b):
   return sum(bin(i).count('1') for i in range(a, b+1))


print(range_bit_count(2,7) , 11)
print(range_bit_count(0,1) , 1)
print(range_bit_count(4,4) , 1)